packages/socket-mode/src/SocketModeClient.ts

```diff
@@ -229,9 +229,13 @@ export class SocketModeClient extends EventEmitter {
     this.logger.debug(`Before trying to reconnect, this client will wait for ${msBeforeRetry} milliseconds`);
     return new Promise((res, _rej) => {
       setTimeout(() => {
-        this.logger.debug('Continuing with reconnect...');
-        this.emit(State.Reconnecting);
-        cb.apply(this).then(res);
+        if (this.shuttingDown) {
```

Comment:
While not a problem brought up by users, I noticed a problem here when writing integration tests:

- if you turn on the socket mode client and it is seeing errors from the web socket server, and
- in between reconnect attempts you call `disconnect()` to stop the client,

.. then the client will keep trying to reconnect forever via this part of the code. Adding this guard here just in case.

Reply:
`this` makes for an awesome catch 🙏 ✨  Thanks for guarding against it!